src/Util/Configuration.php

```diff
@@ -1242,4 +1182,25 @@ private function parseGroupConfiguration(string $root): array
 
         return $groups;
     }
+
+    private function setResult($element): array
```

Comment:
The name `setResult()` does not "speak to me"; it is not clear what the method does. Can we find a better name?

The `$element` parameter should have a type declaration. `DOMElement` looks like the appropriate type, then we can also get rid of the `instanceof` on the first line of the method's body.